j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt

```diff
@@ -49,104 +49,153 @@ class JKClassImpl(override var modifierList: JKModifierList, override var name:
     override fun <R, D> accept(visitor: JKVisitor<R, D>, data: D): R = visitor.visitValueArgument(this, data)
+    override fun <D> transformChildren(transformer: JKTransformer<D>, data: D) {
+        type = type.transform(transformer, data)
+    }
 }
 
 class JKBlockImpl(override var statements: List<JKStatement>) : JKBlock, JKElementBase() {
     override fun <R, D> accept(visitor: JKVisitor<R, D>, data: D): R = visitor.visitBlock(this, data)
+    override fun <D> transformChildren(transformer: JKTransformer<D>, data: D) {
+        statements = statements.map { it.transform<JKStatement, D>(transformer, data) }
+    }
 }
 
 class JKStringLiteralExpressionImpl(override val text: String) : JKStringLiteralExpression, JKElementBase() {
     override fun <R, D> accept(visitor: JKVisitor<R, D>, data: D): R = visitor.visitStringLiteralExpression(this, data)
 }
 
-class JKBinaryExpressionImpl(override val left: JKExpression, override val right: JKExpression?,
-                             override val operator: JKOperatorIdentifier) : JKBinaryExpression, JKElementBase() {
+class JKBinaryExpressionImpl(override var left: JKExpression, override var right: JKExpression?,
+                             override var operator: JKOperatorIdentifier) : JKBinaryExpression, JKElementBase() {
     override fun <R, D> accept(visitor: JKVisitor<R, D>, data: D): R = visitor.visitBinaryExpression(this, data)
     override fun <D> acceptChildren(visitor: JKVisitor<Unit, D>, data: D) {
         left.accept(visitor, data)
         operator.accept(visitor, data)
         right?.accept(visitor, data)
     }
+
+    override fun <D> transformChildren(transformer: JKTransformer<D>, data: D) {
+        left = left.transform(transformer, data)
+        right = right?.transform(transformer, data)
+        operator = operator.transform(transformer, data)
+    }
 }
 
-class JKPrefixExpressionImpl(override val expression: JKExpression?,
-                             override val operator: JKOperatorIdentifier) : JKPrefixExpression, JKElementBase() {
+class JKPrefixExpressionImpl(override var expression: JKExpression?,
+                             override var operator: JKOperatorIdentifier) : JKPrefixExpression, JKElementBase() {
     override fun <R, D> accept(visitor: JKVisitor<R, D>, data: D): R = visitor.visitPrefixExpression(this, data)
     override fun <D> acceptChildren(visitor: JKVisitor<Unit, D>, data: D) {
         expression?.accept(visitor, data)
         operator.accept(visitor, data)
     }
+
+    override fun <D> transformChildren(transformer: JKTransformer<D>, data: D) {
+        expression = expression?.transform(transformer, data)
+        operator = operator.transform(transformer, data)
+    }
 }
 
-class JKPostfixExpressionImpl(override val expression: JKExpression,
-                              override val operator: JKOperatorIdentifier) : JKPostfixExpression, JKElementBase() {
+class JKPostfixExpressionImpl(override var expression: JKExpression?,
```

Comment:
Why made `expression` nullable?

Reply:
OK, after discussion with @Dimach, we decided, that it shouldn't be nullable here, and in `JKPostfixExpression`, but should to in `JKUnaryExpression`